File: LeetCode/Merge Sorted Array/Merge Sorted Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        //take an array of size m+n to merge the two sorted arrays
        int arr[m+n];
        int i=0,j=0,k=0;
        //loop over both the arrays and check for the smaller elements simultaneously
        while(i<m && j<n){
            if(nums1[i]<nums2[j])
                arr[k++]=nums1[i++]; //insert the smaller number into the array arr
            else
                arr[k++]=nums2[j++];
        while(i<m) //copy any leftover elements
            arr[k++]=nums1[i++];
        while(j<n) //copy any leftover elements
            arr[k++]=nums2[j++];
        for(k=0;k<(m+n);k++) //copy the arr array elements into nums1
            nums1[k]=arr[k];
        return;
    }
};

File: LeetCode/Merge Sorted Array/Merge Sorted Array.java
class Solution {
public void merge(int[] nums1, int m, int[] nums2, int n) {

    //3 pointers {i for nums1 , j for nums2, k for output array}
    int i = 0 , j = 0 , k = 0;
    int[] output = new int[m+n];
    
    //run the loop till both the pointers meet the given size of array
    while(i < m && j < n){
        if(nums1[i]<=nums2[j]){
            output[k] = nums1[i];
            i++;
        }else{
            output[k] = nums2[j];
            j++;
        }
        k++;
    }
    
    //if i didn't meet end put all remaining elements of nums1 
    while(i < m){
        output[k]= nums1[i];
        i++;
        k++;
    }
    //if j didn't meet end put all remaining elements of nums2
    while(j < n){
        output[k]= nums2[j];
        j++;
        k++;
    }
    
    //updating values of nums1 array with output array's values 
    for(int idx = 0; idx < nums1.length; idx++){
        nums1[idx] = output[idx];
    }
}
}
